tfx_addons/sampler/executor.py

```diff
@@ -0,0 +1,259 @@
+# Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Executor for TensorFlow Transform."""
+
+import os
+import random
+from typing import Any, Dict, List, Text
+
+import apache_beam as beam
+import tensorflow as tf
+from tfx import types
+from tfx.dsl.components.base import base_beam_executor
+from tfx.dsl.io import fileio
+from tfx.types import artifact_utils
+from tfx.utils import io_utils, json_utils
+
+from tfx_addons.sampler import spec
+
+
+class Executor(base_beam_executor.BaseBeamExecutor):
+  """Executor for Sampler."""
+  def _CreatePipeline(self, unused_transform_output: Text) -> beam.Pipeline:
+    """Creates beam pipeline.
+    Args:
+      unused_transform_output: unused.
+    Returns:
+      Beam pipeline.
+    """
+
+    return self._make_beam_pipeline()
+
+  def Do(
+      self,
+      input_dict: Dict[Text, List[types.Artifact]],
+      output_dict: Dict[Text, List[types.Artifact]],
+      exec_properties: Dict[Text, Any],
+  ) -> None:
+    """Sampler executor entrypoint.
+    Args:
+      input_dict: Input dict from input key to a list of artifacts, including:
+      - examples: A list of type `standard_artifacts.Examples` which should
+      contain custom splits specified in splits_config. If custom split is
+      not provided, this should contain two splits 'train' and 'eval'.
+      output_dict: Output dict from key to a list of artifacts, including:
+      - sampled_examples: sampled examples, only for the given
+      splits as specified in splits. May also include copies of the
+      other non-sampled spits, as specified by null_classes.
+      exec_properties: A dict of execution properties, including:
+      - name: Optional unique name. Necessary if multiple components are
+      declared in the same pipeline.
+      - label: The name of the column containing class names to sample by.
+      - splits: A list containing splits to sample. Defaults to ['train'].
```

Comment:
This default of ['train'] doesn't appear to be enforced in code. If you don't put in a `splits` parameter your component will fail.

Reply:
Sorry, the documentation wasn't updated - setting the default to `['train']` gets me a linter error, so I set the default to None and raised an error for it. Will update the documentation.